test_pull_requests

```diff
@@ -461,31 +413,107 @@ class FlushableStore
   end
 end
 
-# Quick-and-dirty middleware that attempts to invalidate appropriate
-# cache entries before handing the request off to the
-# Faraday::HttpCache middleware
-class GitHubCacheInvalidator < Faraday::Middleware
 
-  UNSAFE_METHODS = [:post, :put, :delete, :patch].freeze
+class InvalidatingGitHubCache < Faraday::HttpCache
+
+  class CustomInvalidatingStorage < Faraday::HttpCache::Storage
+
+    attr_accessor :invalidatable
+
+    def initialize(*args)
+      super
+      @invalidatable = Hash.new { |h,k| h[k] = {} }
+      @cache.storage = self
+    end
+
+    # GitHub-specific (although probably other APIs are similar) method
+    # to remove the pagination parameters from a URL so it can be better
+    # classified
+    #
+    # Example:
+    #   synthesize_url("https://api.github.com/some/resource/id?per_page=100")
+    #     returns
+    #   "https://api.github.com/some/resource/id"
+    def synthesize_url(orig_url)
+      url = orig_url.dup
+      uri = Addressable::URI.parse url
+      params = uri.query_values
+      if params
+        params.delete 'per_page'
+        params.delete 'page'
+        uri.query_values = params
+        url = uri.to_s.chomp('?')
+        # get rid of trailing numeric ID in REST path;
+        # e.g. '/issues/comments/311375581' -> '/issues/comments'
```

Comment:
the GitHub API call to delete a comment from an issue/pr takes a comment ID that is global to the repo, but it changes the response for the call to get comments on a particular issue. Since we can't tell which issue was impacted, we need to invalidate all the issue comments we have cached for the repo.